multiversx_sdk/network_providers/proxy_network_provider.py

```diff
@@ -178,10 +179,22 @@ def await_transaction_on_condition(self,
 
         return awaiter.await_on_condition(tx_hash, condition)
 
-    def query_contract(self, query: IContractQuery) -> ContractQueryResponse:
-        request = ContractQueryRequest(query).to_http_request()
+    def query_contract(self, query: SmartContractQuery) -> SmartContractQueryResponse:
+        query_converter = SmartContractQueryConverter()
+        request = query_converter.smart_contract_query_to_dictionary(query)
         response = self.do_post_generic('vm-values/query', request)
-        return ContractQueryResponse.from_http_response(response.get('data', ''))
+        response = response.get('data', '')
+
+        return_data = response.get('returnData', []) or response.get('ReturnData', [])
+        return_code = response.get('returnCode', '') or response.get('ReturnCode', '')
+        return_message = response.get('returnMessage', '') or response.get('ReturnMessage', '')
```

Comment:
Now I see. Fine for now, let's think of other ways to tackle this (in the background) :+1: 

Our converter for `SmartContractQuery` has knowledge of protocol-desired JSON-ifying (which seems fine in practice, although it somehow promises that it converts the query to a dictionary, not necessarily to a protocol-desired dictionary).

On the other hand, here we have a way of converting the bare HTTP response payload to our DTO, the `SmartContractQueryResponse`.

Thus, there's a design asymmetry. 

Can stay as it is for now.

Reply:
As discussed, will do this in the next PR.